commit 9e395fcbe3ee7b99d737d67f183ce1880f184e22
Author: sveinung <sveinung@a0f10bec-cc02-0410-94fc-a9cfff90b4cd>
Date:   Thu Jun 30 06:19:03 2016 +0000

    protocol: send action range in ruleset packet.
    
    Transfer the minimum and the maximum distance between actor and target in
    PACKET_RULESET_ACTION. This makes the range available to clients that don't
    use Freeciv's C code.
    
    See patch #7348
    
    git-svn-id: svn://svn.gna.org/svn/freeciv/trunk@33114 a0f10bec-cc02-0410-94fc-a9cfff90b4cd

diff --git a/client/packhand.c b/client/packhand.c
index d7f3e61..34c9ae5 100644
--- a/client/packhand.c
+++ b/client/packhand.c
@@ -3933,5 +3933,8 @@ void handle_ruleset_action(const struct packet_ruleset_action *p)
 
   sz_strlcpy(act->ui_name, p->ui_name);
   act->quiet = p->quiet;
+
+  act->min_distance = p->min_distance;
+  act->max_distance = p->max_distance;
 }
 
diff --git a/common/packets.def b/common/packets.def
index 2844d88..c3f457f 100644
--- a/common/packets.def
+++ b/common/packets.def
@@ -1716,5 +1716,8 @@ PACKET_RULESET_ACTION = 246; sc, lsend
 
   STRING ui_name[MAX_LEN_NAME];
   BOOL   quiet;
+
+  SINT32 min_distance;
+  SINT32 max_distance;
 end
 
diff --git a/server/ruleset.c b/server/ruleset.c
index 4724c9f..9f8a23b 100644
--- a/server/ruleset.c
+++ b/server/ruleset.c
@@ -7135,5 +7135,8 @@ static void send_ruleset_actions(struct conn_list *dest)
     packet.id = act;
     sz_strlcpy(packet.ui_name, action_by_number(act)->ui_name);
     packet.quiet = action_by_number(act)->quiet;
+
+    packet.min_distance = action_by_number(act)->min_distance;
+    packet.max_distance = action_by_number(act)->max_distance;
 
     lsend_packet_ruleset_action(dest, &packet);
